core/kernel/ldelf_loader.c

```diff
@@ -356,26 +371,24 @@ TEE_Result ldelf_dlopen(struct user_mode_ctx *uctx, TEE_UUID *uuid,
 	uint32_t panic_code = 0;
 	uint32_t panicked = 0;
 	struct ts_session *sess = NULL;
+	struct dl_entry_arg dl_entry_arg = { };
 
 	assert(uuid);
 
 	usr_stack -= ROUNDUP(sizeof(*arg), STACK_ALIGNMENT);
 	arg = (struct dl_entry_arg *)usr_stack;
 
-	res = vm_check_access_rights(uctx,
-				     TEE_MEMORY_ACCESS_READ |
-				     TEE_MEMORY_ACCESS_WRITE |
-				     TEE_MEMORY_ACCESS_ANY_OWNER,
-				     (uaddr_t)arg, sizeof(*arg));
-	if (res) {
-		EMSG("ldelf stack is inaccessible!");
+	dl_entry_arg.cmd = LDELF_DL_ENTRY_DLOPEN;
+	dl_entry_arg.dlopen.uuid = *uuid;
+	dl_entry_arg.dlopen.flags = flags;
+
+	res = clear_user(arg, sizeof(*arg));
+	if (res)
```

Comment:
is it really needed to clear `*arg` since the full `*arg` content is loaded by the next instruction?

Reply:
oh yeah. I removed `clear_user()` call